### PR TITLE
Add inference utils

### PR DESCRIPTION
**Pull request by yeandy**
Adding inference utils. PTAL @mailvijayasingh @entrpn

### CODE REVIEW

File: src/maxdiffusion/utils/inference_utils.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+import jax.numpy as jnp
+import jax
+import numpy as np
+
+from maxdiffusion import max_logging
+
+FrozenDict = core.frozen_dict.FrozenDict
+
+
+def loop_body(
```
Comment (entrpn) [inline src/maxdiffusion/utils/inference_utils.py:38]: a lot of these functions seem to be copied over from generate_sdxl.py. Can we refactor generate_sdxl so we don't have duplicate code in two files?

File: src/maxdiffusion/utils/inference_utils.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        return_tensors="np",
+    )
+    inputs.append(text_inputs.input_ids)
+  inputs = jnp.stack(inputs, axis=1)
+  return inputs
+
+
+def tokenize_one(prompt, tokenizer):
+  inputs = []
+  # for _tokenizer in [pipeline.tokenizer, pipeline.tokenizer_2]:
```
Comment (entrpn) [inline src/maxdiffusion/utils/inference_utils.py:118]: remove commented code.

File: src/maxdiffusion/utils/inference_utils.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+def vae_decode(latents, state, pipeline):
+  latents = 1 / pipeline.vae.config.scaling_factor * latents
+  image = pipeline.vae.apply(
+      {"params": state.params}, latents, method=pipeline.vae.decode
+  ).sample
+  image = (image / 2 + 0.5).clip(0, 1).transpose(0, 2, 3, 1)
+  return image
+
+
+def create_device_mesh(config, devices=None, logging=True):
```
Comment (entrpn) [inline src/maxdiffusion/utils/inference_utils.py:138]: same as above, this is duplicated code from max_utils. Can we max_utils functions as imports.

**Comment by yeandy** [comment]
Thanks @entrpn. Synced with Vijaya, and I think we actually don't need a merge right now; just this `inference_utils` branch to use for reference.

**Pull request closed without merging**